### PR TITLE
Don't write aAnsi when it's not filled (fixes #5340)

### LINKED ISSUE #5340: Something is up with the server log
```
uQH[2022-06-05 10:46:26][server]: server name is 'unnamed server'
etQH[2022-06-05 10:46:26][datafile]: loading data index=25 size=1658 uncompressed=423660
etQH[2022-06-05 10:46:26][datafile]: loading data index=26 size=1025 uncompressed=484320
etQH[2022-06-05 10:46:26][datafile]: loading data index=27 size=5464 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=28 size=6281 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=29 size=9001 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=30 size=5633 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=31 size=4108 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=32 size=21698 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=33 size=4542 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=34 size=9822 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=35 size=5360 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=36 size=3979 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=37 size=6465 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=38 size=4454 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=39 size=19821 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=40 size=27573 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=41 size=3972 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=42 size=4019 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=43 size=3952 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=45 size=3952 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=47 size=3952 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=49 size=6280 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=50 size=3962 uncompressed=4054900
etQH[2022-06-05 10:46:26][datafile]: loading data index=51 size=3952 uncompressed=4054900
tQH[2022-06-05 10:46:26][datafile]: loading data index=48 size=2010 uncompressed=2027450
tQH[2022-06-05 10:46:26][datafile]: loading data index=44 size=6102 uncompressed=6082350
tQH[2022-06-05 10:46:26][datafile]: loading data index=46 size=4021 uncompressed=4054900
UrQH[2022-06-05 10:46:26][datafile]: loading data index=3 size=86 uncompressed=101
PH[2022-06-05 10:46:26][chatresp]: No such command: sv_jailstate.
PH[2022-06-05 10:46:26][chatresp]: No such command: sv_spawntiles_mode.
PH[2022-06-05 10:46:26][chatresp]: No such command: sv_pvp_arena_state.
PH[2022-06-05 10:46:26][chatresp]: No such command: sv_dummy_block_points.
tQH[2022-06-05 10:46:26][git-revision]: b554039740e7c5ed
uQH[2022-06-05 10:46:26][server]: version 16.1 on linux amd64
uQH[2022-06-05 10:46:26][server]: git revision hash: b554039740e7c5ed
UQH[2022-06-05 10:46:26][register/6/ipv6]: registering...
UQH[2022-06-05 10:46:26][register/6/ipv4]: registering...
UQH[2022-06-05 10:46:26][register/7/ipv6]: registering...
QH[2022-06-05 10:46:26][register/7/ipv4]: registering...
Z([2022-06-05 10:46:26][sql]: load best time done on read database 0
```

Note the prefixed chunk

### PR DESCRIPTION
**Pull request by def-**
## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by ChillerDragon** [comment]
Tested. And works! Thanks :heart:

**Comment by ChillerDragon** [comment]
bors +r

**Comment by C0D3D3V** [comment]
Thanks for the quick fix :)
bors r+

File: src/base/log.cpp [hunk 1]
```diff
@@ -234,13 +234,14 @@ class CLoggerAsync : public ILogger
...
+				aio_write_unlocked(m_pAio, aAnsi, str_length(aAnsi));
 			}
-			aio_write_unlocked(m_pAio, aAnsi, str_length(aAnsi));
 		}
 		aio_write_unlocked(m_pAio, pMessage->m_aLine, pMessage->m_LineLength);
 		if(m_AnsiTruecolor && pMessage->m_HaveColor)
 		{
-			aio_write_unlocked(m_pAio, "\x1b[0m", 4); // reset
+			const char aResetColor[] = "\x1b[0m";
+			aio_write_unlocked(m_pAio, aResetColor, sizeof(aResetColor)); // reset
```
Comment (heinrich5991) [inline src/base/log.cpp:244]: This writes a null byte.
```suggestion
			aio_write_unlocked(m_pAio, aResetColor, str_length(aResetColor)); // reset
```

**Pull request merged**